runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -408,25 +409,47 @@ void chpl_task_callMain(void (*chpl_main)(void)) {
   // since we want to run all work in a task with a comm-friendly stack,
   // run main in a pthread that we will wait for.
   size_t stack_size;
-  size_t align;
   pthread_attr_t attr;
   pthread_t thread;
-  void* stack;
   int rc;
+#ifndef CHPL_USING_CSTDLIB_MALLOC
+  size_t align;
+  int free_flag;
+  void* stack;
+#endif
+
+  rc = pthread_attr_init(&attr);
+  if( rc != 0 ) {
+    chpl_internal_error("pthread_attr_init main failed");
+  }
 
   stack_size  = chpl_thread_getCallStackSize();
+
+/* Uses a registered stack, if possible. For the reason about
+ * this define, see threads-pthreads.c
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
   align = chpl_getSysPageSize();
-  stack = chpl_memalign(align, stack_size);
+  stack = chpl_memalign(align, stack_size + align);
```

Comment:
Because stack_size is the size set by the user. I need to allocate enough space for the stack and the extra memory guard page, so I allocate stack_size + align. I can't mprotect an area inside [stack, stack + stack_size] because the standard [[1](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_attr_getstack.html)] says "All pages within the stack described by stackaddr and stacksize shall be both readable and writable by the thread."